SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -136,6 +144,102 @@ void FLocalDeploymentManager::RefreshServiceStatus()
 	});
 }
 
+bool FLocalDeploymentManager::CheckIfPortIsBound(int32 Port)
+{
+	ISocketSubsystem* SocketSubsystem = ISocketSubsystem::Get(PLATFORM_SOCKETSUBSYSTEM);
+	bool bSuccess = false;
+	// Set our broadcast address
+	TSharedRef<FInternetAddr> BroadcastAddr = SocketSubsystem->CreateInternetAddr();
+	BroadcastAddr->SetBroadcastAddress();
+	BroadcastAddr->SetPort(Port);
+	// Now the listen address
+	TSharedRef<FInternetAddr> ListenAddr = SocketSubsystem->GetLocalBindAddr(*GLog);
+	ListenAddr->SetPort(Port);
+	// Now create and set up our sockets
+	FSocket* ListenSocket = SocketSubsystem->CreateSocket(NAME_Stream, TEXT("LAN beacon"), false);
+	if (ListenSocket != NULL)
+	{
+		ListenSocket->SetReuseAddr();
+		ListenSocket->SetNonBlocking();
+		ListenSocket->SetRecvErr();
+		// Bind to our listen port
+		if (ListenSocket->Bind(*ListenAddr))
+		{
+			bSuccess = ListenSocket->Listen(30);
+		}
+		else
+		{
+			UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to bind listen socket to addr (%s) for LAN beacon"), *ListenAddr->ToString(true));
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to create listen socket for LAN beacon"));
+	}
+	bool bSuccessfullyBoundPort = bSuccess && ListenSocket;
+	if (ListenSocket != NULL) {
+		// Kill the blocking thing we just opened.
+		ListenSocket->Close();
+	}
+	/* Either we couldn't create the socket or couldn't listen on it so the port is probably not bound */
+	return !bSuccessfullyBoundPort;
+}
+
+bool FLocalDeploymentManager::PreStartCheck()
+{
+	const int runtimePort = 5301;
```

Comment:
```suggestion
	const int RuntimePort = 5301;
```
Might want to change this to a static const somewhere.